### DISCUSSION TITLE
How to run LightningCLI in Kaggle Notebook

### DISCUSSION THREAD

**Discussion by xiaoyuan0203** [Lightning-AI/pytorch-lightning#13277] [General]
How to run LightningCLI in Kaggle Notebook? 

![image](https://user-images.githubusercontent.com/78028879/173280556-373e39cd-7838-4bf7-9430-d0ec663f1f0b.png)

![image](https://user-images.githubusercontent.com/78028879/173280660-8fc74dde-5436-4ddb-b0ff-216cfe17c651.png)

**Comment by akihironitta** [comment]
@xiaoyuan0203 I'm not an expert of the Kaggle Notebook environment, but I don't think it's feasible to run your code with LightningCLI since you can't call your script from command-line... Am I missing something?

**Reply by xiaoyuan0203** [reply]
Maybe you are right. Another question! I am considering how to give value to the arguments in the script (not in the command-line) before calling `cli.trainer`, `cli.model` while using LightningCLI

**Reply by akihironitta** [reply]
> how to give value to the arguments in the script

What kind of arguments specifically? (trainer args, model args, other custom args, etc.)

---

By the way, it's always a good idea to ask something about a different topic in a different discussion... Could you create another discussion for your further unrelated questions?

**Comment by mauvilsa** [comment]
There is now an official way to do this, see https://pytorch-lightning.readthedocs.io/en/latest/cli/lightning_cli_advanced_3.html#run-from-python
